<launch>

    <node pkg="audio_utils" exec="capture_node" name="audio_capture" output="screen">
        <param name="backend" value="alsa"/>
        <param name="device" value="default"/>
        <param name="format" value="signed_16"/>
        <param name="channel_count" value="1"/>
        <param name="sampling_frequency" value="16000"/>
        <param name="frame_sample_count" value="512"/>
        <param name="merge" value="false" />
        <param name="gain" value="1.0" />

        <remap from="audio_out" to="audio_input_signed_16_16000"/>
    </node>

    <node pkg="speech_to_text" exec="google_speech_to_text_node.py" name="google_speech_to_text_node" output="screen">
        <param name="sampling_frequency" value="16000"/>
        <param name="frame_sample_count" value="512"/>
        <param name="request_frame_count" value="5"/>
        <param name="language" value="en"/>

        <remap from="audio_in" to="audio_input_signed_16_16000"/>
        <remap from="audio_in/filter_state" to="speech_to_text/filter_state"/>
    </node>

</launch>
